src/WebJobs.Script/Host/DebugStateProvider.cs

```diff
@@ -12,45 +12,55 @@ namespace Microsoft.Azure.WebJobs.Script
     internal class DebugStateProvider : IDebugStateProvider, IDisposable
     {
         internal const int DebugModeTimeoutMinutes = 15;
+        internal const int DiagnosticModeTimeoutHours = 2;
```

Comment:
Good starting point. It'd be cool if we could parse the file and let this be configurable (if we want to let someone run overnight -- or just for a few minutes -- with diagnostics on, for example). But that seems completely additive if we find we need it.

Reply:
Yes we can add that later if needed. I will increase this to 3 hrs though.